scipy/stats/_fit.py

```diff
@@ -688,3 +689,546 @@ def nllf(free_params, data=data):  # bind data NOW
+        rfd_vals = fit_fun(data)
+        rfd_dist = dist(*rfd_vals)
+        return compare_fun(rfd_dist, data)
+
+    res = stats.monte_carlo_test(data, rvs, statistic_fun, vectorized=True,
+                                 n_resamples=n_mc_samples, axis=-1,
+                                 alternative='greater')
+    opt_res = optimize.OptimizeResult()
+    opt_res.success = True
+    opt_res.message = "The fit was performed successfully."
+    opt_res.x = nhd_vals
+    # Only continuous distributions for now, hence discrete=False
+    # There's no fundamental limitation; it's just that we're not using
+    # stats.fit, discrete distributions don't have `fit` method, and
+    # we haven't written any vectorized fit functions for a discrete
+    # distribution yet.
+    return GoodnessOfFitResult(FitResult(dist, data, False, opt_res),
+                               res.statistic, res.pvalue,
+                               res.null_distribution)
+
+
+def _get_fit_fun(dist, data, fit_method, guessed_params, fixed_params):
+
+    shape_names = [] if dist.shapes is None else dist.shapes.split(", ")
+    param_names = shape_names + ['loc', 'scale']
+    fparam_names = ['f'+name for name in param_names]
+    all_fixed = not set(fparam_names).difference(fixed_params)
+    guessed_shapes = [guessed_params.pop(x, None)
+                      for x in shape_names if x in guessed_params]
+
+    # Define statistic, including fitting distribution to data
+    if dist in _fit_funs:
+        def fit_fun(data):
+            params = _fit_funs[dist](data, **fixed_params,
+                                     fit_method=fit_method)
+            params = np.asarray(np.broadcast_arrays(*params))
+            if params.ndim > 1:
+                params = params[..., np.newaxis]
+            return params
+
+    elif all_fixed:
+        def fit_fun(data):
+            return [fixed_params[name] for name in fparam_names]
+
+    else:
+        def fit_fun_1d(data):
+            return dist.fit(data, *guessed_shapes, **guessed_params,
+                            **fixed_params, method=fit_method)
+
+        def fit_fun(data):
+            params = np.apply_along_axis(fit_fun_1d, axis=-1, arr=data)
+            if params.ndim > 1:
+                params = params.T[..., np.newaxis]
+            return params
+
+    return fit_fun
+
+
+# Vectorized fitting functions. These are to accept ND `data` in which each
+# row (slice along last axis) is a sample to fit, scalar fixed parameters,
+# and `fit_method`. They return a tuple of shape parameter arrays, each of
+# shape data.shape[:-1].
+def _fit_norm(data, floc=None, fscale=None, fit_method="mle"):
+    loc = floc
+    scale = fscale
+    if loc is None and scale is None:
+        loc = np.mean(data, axis=-1)
+        scale = np.std(data, ddof=1, axis=-1)
+    elif loc is None:
+        loc = np.mean(data, axis=-1)
+    elif scale is None:
+        scale = np.sqrt(((data - loc)**2).mean(axis=-1))
```

Comment:
From [[1]](https://www.jstor.org/stable/2286009):

![image](https://user-images.githubusercontent.com/6570539/195667558-4aec4e5a-bb56-4ea5-a0a8-1089a79be4a9.png)

This is Case 2.

The estimate of variance with $n$ is unbiased when the mean is known ([here](https://stats.stackexchange.com/questions/162842/estimate-variance-of-a-population-if-population-mean-is-known)).

Actually, one could argue that using $n-1$ (even when location is unknown) is incorrect if we say we're using maximum likelihood estimation. The maximum likelihood estimate for scale is with $n$ (not $n-1$) whether the location is known or not. Anderson suggested using this bias-corrected fit and calculated critical values accordingly, but we're calculating critical values on the fly, so it doesn't really matter how we do the fit as long as we stay consistent. 

With this in mind, I think I'll remove the `method` parameter, and I'll explain in the notes that we use MLE but sometimes with bias correction. I don't think we want to expose MM as an option because it's rarely preferable.

Reply:
thanks for the clarification. do you still plan to update the docs?

Reply:
Yes, I'm removing `fit_method`.